numba/targets/arraymath.py

```diff
@@ -2344,6 +2344,55 @@ def np_imag_impl(a):
 #----------------------------------------------------------------------------
 # Misc functions
 
+@overload(np.delete)
+def np_delete(arr, obj):
+    # Implementation based on numpy
+    # https://github.com/numpy/numpy/blob/af66e487a57bfd4850f4306e3b85d1dac3c70412/numpy/lib/function_base.py#L4065-L4267
+
+    if not isinstance(arr, (types.Array, types.Sequence)):
+        raise TypingError("arr must be either an Array or a Sequence")
+
+    def np_delete_impl(arr, obj):
+        arr = np.ravel(np.asarray(arr))
+        N = arr.size
+
+        keep = np.ones(N, dtype=np.bool_)
+        obj = np.array(obj)
+        keep[obj] = False
+        return arr[keep]
+
+    def np_delete_scalar_impl(arr, obj):
+        arr = np.ravel(np.asarray(arr))
+        N = arr.size
+        pos = obj
+
+        if (pos < -N or pos >= N):
+            raise IndexError('pos must be less than the len(arr)')
+            # NumPy raises IndexError: index 'i' is out of
+            # bounds for axis 'x' with size 'n'
+
+        if (pos < 0):
+            pos += N
+
+        return np.concatenate((arr[:pos], arr[pos+1:]))
+
+    def np_delete_slice_impl(arr, obj):
```

Comment:
You can get around this sort of "function is __almost__ identical" duplication via methods like this: https://github.com/stuartarchibald/numba/commit/194cd6d3ab97b92ebd2534ae852672ae2af7b5c4 (which you are welcome to use). `@register_jitable` is just like `@jit` but for internal use, the LLVM inliner pretty much always inlines these trivial function variations so there's no call penalty.